superset-frontend/src/dashboard/actions/nativeFilters.ts

```diff
@@ -142,51 +138,49 @@ export const setFilterSetsConfiguration = (
   }
 };
 
-export const SET_EXTRA_FORM_DATA = 'SET_EXTRA_FORM_DATA';
-export interface SetExtraFormData {
-  type: typeof SET_EXTRA_FORM_DATA;
+export const UPDATE_EXTRA_FORM_DATA = 'UPDATE_EXTRA_FORM_DATA';
+export interface UpdateExtraFormData {
+  type: typeof UPDATE_EXTRA_FORM_DATA;
   filterId: string;
-  extraFormData: ExtraFormData;
-  currentState: CurrentFilterState;
+  nativeFilters?: Omit<FilterState, 'id'>;
+  crossFilters?: Omit<FilterState, 'id'>;
+  ownFilters?: Omit<FilterState, 'id'>;
```

Comment:
Would it make more sense to make `id` optional on `FilterState`?

Reply:
this type we using for redux structure, so we must have `id` there, but for action we don't need it, so I preferred to use `Omit` instead create new type, we already have a lot of similar types

Reply:
Ok, makes sense. Alternatively we might extend to something like `FilterStateWithId`, but that would probably be overkill 🙂

Reply:
I also think so :)